File: 02Fabryka/Students/2017/FactoryMethod/src/design/patterns/store/BookStore.java
package design.patterns.store;

import design.patterns.books.Book;
import design.patterns.books.enums.BookType;

public abstract class BookStore {

    public Book order(BookType bookType) {
        Book book = createBook(bookType);
        book.collecting();
        book.packing();
        book.send();

        return book;
    }

    protected abstract Book createBook(BookType bookType);
}

File: 02Fabryka/Students/2018/OlszerKrzysztof/Factory/FactoryMethod/Books/BookStore.cs
using System;

namespace FactoryMethod.Books
{
    public abstract class BookStore
    {
        public abstract Book createBook(string id);
        public Book read(string id)
        {
            Book book = createBook(id);
            String res = book.open();
            res += book.getName();
            res += book.getNoPages();
            Console.WriteLine(res);
            return book;
        }
    }
}
